mirage/client/dns_client_mirage.ml

```diff
@@ -310,4 +367,19 @@ module Make (R : Mirage_random.S) (T : Mirage_time.S) (M : Mirage_clock.MCLOCK)
   end
 
   include Dns_client.Make(Transport)
+
+  let connect_device ?(protocol= "udp") ?(authenticators= []) ?(nameservers= []) ?(timeout= 5_000_000L) stack =
```

Comment:
- UDP shouldn;t be the default
- please use Duration instead of 5_000_000L
- the interface is rather brittle, since it allows multiple nameservers and authenticators, but they are unrelated. also they need to use the same protocol.

What about `connect_device : <> -> ?timeout -> stack -> t` where the `<>` is an (optional) list of:
- protocol (udp, tcp, tls)
- ip address
- port (optional)
- authenticator (if tls was used)

The authenticator may as well need to allow to pass the hostname to validate against (instead of using the IP address).

Reply:
I don't really understand the format and I just would like to clarify a point: we are restricted by what `Mirage.Key.Arg` can give to us (because this `connect_device` will be used then by `functoria`) and we don't many possibilities (only `string`, `ipaddr` or `int`).

So can you elaborate what you expect about:
> What about connect_device : <> -> ?timeout -> stack -> t where the <> is an (optional) list of:

Reply:
The issue I see with the current signature: `?protocol -> ?authenticators -> ?nameservers` is that there is a single protocol that all nameservers use, and also there's only loosely coupled "authenticators" with "nameservers".

I'd instead be in favour of having a single `string` that uses can pass that includes all information about a resolver:
- `PROTO:IP:PORT:AUTH`

So, as an example:
- `141.1.1.1` (default protocol, default port)
- `UDP:141.1.1.1` (UDP, default port)
- `TLS:8.8.8.8:key-fp:aaaaa` (TLS, default port, pinned to a specific public key)

In this way, a user can provide multiple name servers with distinct protocols and tightly coupled authenticators.

As noted earlier, and again above, the `protocol` (in respect to the dns-client dns-client-mirage interaction) is always `` `Tcp `` (since udp is not supported, and tls uses the same format as tcp).

Reply:
Ok I see, I will come back with something better.

Reply:
Thanks @dinosaure. This PR is highly appreciated (to finally use the dns-client directly in MirageOS unikernels), and pretty close for being merged and released. Sorry for my recent lack of time on reviewing this PR (and thanks for taking the time to push the changes to x509/ipaddr/...) :D